core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -286,6 +286,32 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
+  override def completeTasks(partitionId: Int, stageId: Int, killTasks: Boolean): Unit = {
+    taskSetsByStageIdAndAttempt.getOrElse(stageId, Map()).values.foreach { tsm =>
+      tsm.partitionToIndex.get(partitionId) match {
+        case Some(index) =>
+          val taskInfoList = tsm.taskAttempts(index)
+          taskInfoList.foreach { taskInfo =>
+            tsm.markPartitionCompleted(partitionId, taskInfo)
```

Comment:
why do we need to call it for all the attempts?

Reply:
Yeah good point, will only call it for those tasks for whom `taskInfo.successful` is true. Will update the code shortly. Thank you.

Reply:
Have updated the code to call it only for the one task that succeeded thus triggering the event loop in the first place. Thank you.